### ISSUE TITLE
Accessibility keyboard navigation doesn't work for both Leo AI and Search options on the NTP widget.

### ISSUE THREAD

**Issue by GeetaSarvadnya** [brave/brave-browser#54203]
### Description

Found while testing https://github.com/brave/brave-browser/issues/48179
Accessibility keyboard navigation doesn't work for both Leo AI and Search options on the NTP widget.

### Steps to reproduce

1. Clean profile `1.89.128`
2. Enable brave://flags/#brave-ai-chat-show-input-on-new-tab-page
3. Open NTP
4. Ensure that Leo AI and search widget is shown on the NTP
5. Press TAB on Leo AI and search widget via NTP
6. Confirmed that TAB highlights the `query box`, `Leo Tools`, `Attachment` and `Send message to Leo` buttons/text areas as expected
7. Observe that there is no cursor keyboard accessibility for Leo AI and Search options

### Actual result

Accessibility keyboard navigation doesn't work for both Leo AI and Search options on the NTP widget.

https://github.com/user-attachments/assets/865e8e06-1921-4798-8d4f-38b03bff69aa

### Expected result

Accessibility keyboard navigation should work for both Leo AI and Search options on the NTP widget.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave | 1.89.128 Chromium: 147.0.7727.25 (Official Build) beta (arm64)
-- | --
Revision | 39096907d3237d427408388a62dbec5da257544d
OS | macOS Version 15.5 (Build 24F74)

### Channel information

- [ ] release (stable)
- [x] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

cc: @rebron @brave/qa-team

**Comment by zenparsing** [comment]
It appears that both sides of the Search/Chat toggle are keyboard-focusable, but there is no visual indication of focus. This issue also applies to usage of `SegmentedControlItem` in the Brave Wallet WebUI.

https://github.com/brave/leo/issues/1357
